### PR TITLE
feat: metrics for when blocks are submitted [ENG-702]

### PR DESCRIPTION
**Pull request by rswanson**

### CODE REVIEW

**Comment by rswanson** [comment]
* **#33** <a href="https://app.graphite.dev/github/pr/init4tech/builder/33?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/init4tech/builder/33?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/tasks/tx_poller.rs [hunk 1]
```diff
@@ -54,6 +56,7 @@ impl TxPoller {
         self.seen_txns.entry(*tx.tx_hash()).or_insert_with(|| {
             // add to unique transactions
             unique.push(tx.clone());
+            counter!("builder.unique_txs").increment(1);
```
Comment (dylanlott) [inline src/tasks/tx_poller.rs:59]: Should we bucket these by block number maybe?

File: src/tasks/submit.rs [hunk 2]
```diff
@@ -264,12 +266,16 @@ impl SubmitTask {
...
             loop {
                 if let Some(in_progress) = inbound.recv().await {
+                    let building_start_time = Instant::now();
                     let mut retries = 0;
                     loop {
                         match self.handle_inbound(&in_progress).await {
                             Ok(ControlFlow::Retry) => {
                                 retries += 1;
                                 if retries > 3 {
+                                    counter!("builder.building_too_many_retries").increment(1);
```
Comment (dylanlott) [inline src/tasks/submit.rs:276]: Very nice to have for internal visibility 🔥

**Review by dylanlott — APPROVED** [review]
Elapsed block build time is going to be fairly inaccurate because of how few blocks actually are trying to be built. It'll get better once we have some actual traffic on Test Net. For now this is a great start 🎀

**Pull request merged**